File: 231120/줄 세우기 2/line-up-students-2.java
import java.io.*;
import java.util.*;

public class Main {
    public static class Student {
        int index;
        int height;
        int weight;

        public Student(int index, int height, int weight) {
            this.index = index;
            this.height = height;
            this.weight = weight;
        }

        @Override
        public String toString() {
            return String.format("%d %d %d", height, weight, index);
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        List<Student> students = new ArrayList<>();

        int n = Integer.parseInt(br.readLine());
        for (int i = 1; i <= n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int height = Integer.parseInt(st.nextToken());
            int weight = Integer.parseInt(st.nextToken());
            students.add(new Student(i, height, weight));
        }

        Collections.sort(students, (s1, s2) -> {
            if (s1.height != s2.height) {
                return s1.height - s2.height;
            } else {
                return s2.weight - s1.weight;
            }
        });

        for (Student student : students) {
            System.out.println(student.toString());
        }
    }
}

File: 240320/줄 세우기 2/line-up-students-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);

// 학생들의 정보를 나타내는 클래스 선언
class Student {
  constructor(height, weight, number) {
    this.height = height;
    this.weight = weight;
    this.number = number;
  }
}

const students = [];
for (let i = 1; i <= n; i++) {
  const [height, weight] = input[i].split(' ').map(Number);
  students.push(new Student(height, weight, i));
}

// custom comparator를 활용한 정렬
// 키 오름차순, 몸무게 내림차순 순으로 정렬합니다.
students.sort((a, b) => {
  if (a.height === b.height) {
    return b.weight - a.weight; // 몸무게 내림차순
  }
  return a.height - b.height; // 키 오름차순
});

// 결과를 출력합니다.
for (let student of students) {
  console.log(student.height, student.weight, student.number);
}